docs/modules/troubleshoot/pages/error-handling.adoc

```diff
@@ -54,19 +54,12 @@ reprocess their input.
 === Processing Guarantees
 
 Streaming jobs with mutable state, those with a xref:fault-tolerance:fault-tolerance.adoc#processing-guarantee-is-a-shared-concern[processing guarantee]
-set, achieve fault tolerance by periodically saving xref:fault-tolerance:fault-tolerance.adoc#distributed-snapshot[recovery snapshots]. When
-such a job fails, not only does its execution stop but also its
-snapshots get deleted. This makes it impossible to resume it without
-loss.
-
-To cope with this situation, you can configure a job to be suspended in
-the case of a failure, instead of failing completely. For details see
-link:https://docs.hazelcast.org/docs/{full-version}/javadoc/com/hazelcast/jet/config/JobConfig.html#setSuspendOnFailure(boolean)[`JobConfig.setSuspendOnFailure`]
+set, achieve fault tolerance by periodically saving xref:fault-tolerance:fault-tolerance.adoc#distributed-snapshot[recovery snapshots]. If a streaming job was allowed to fail, the snapshots would be deleted. For this reason, by default all streaming jobs are suspended on failure, instead of failing completely. For more details, see link:https://docs.hazelcast.org/docs/{full-version}/javadoc/com/hazelcast/jet/config/JobConfig.html#setSuspendOnFailure(boolean)[`JobConfig.setSuspendOnFailure`]
```

Comment:
interesting fact is that you cannot set `suspendOnFailure` to false for streaming jobs created using SQL `CREATE JOB`

Reply:
I've added a note about this.